beartype_test/a00_unit/a00_typing/test_typing.py

```diff
@@ -107,6 +115,19 @@ def test_api_typing() -> None:
         # Else, this attribute is public and thus unignorable.
     }
 
+    if IS_PYTHON_AT_LEAST_3_7 and _HAZ_TYPING_EXTENSIONS:
+        # Not really "official", but we'll fake it
+        official_typing_attr_name_to_value.update({
+            extension_typing_attr_name: getattr(
+                typing_extensions, extension_typing_attr_name)
+            for extension_typing_attr_name in (
+                'Protocol',
+                'SupportsIndex',
+                'runtime_checkable',
+            )
+
+        })
+
```

Comment:
By definition, your approach is the right approach – because it's the only approach. You're the first to brave these roiling waters. I applaud such bald-faced courage.

Reply:
Well, _that's_ good, because #107 proposes changing it (at least slightly). (Note https://github.com/beartype/beartype/pull/107#issuecomment-1051314978, however.)